edx_rest_framework_extensions/auth/jwt/authentication.py

```diff
@@ -72,31 +73,50 @@ def authenticate(self, request):
             if not user_and_auth:
                 return user_and_auth
 
-            # Not using JWT cookies, CSRF validation not required
-            use_jwt_cookie_requested = request.META.get(USE_JWT_COOKIE_HEADER)
-            if not use_jwt_cookie_requested:
+            # JWT found in authorization header, CSRF validation not required
+            if authorization_header:
                 return user_and_auth
 
+            # Enforce CSRF for JWT Cookie (assumed if not using the authorization header)
             self.enforce_csrf(request)
 
             # CSRF passed validation with authenticated user
             return user_and_auth
 
-        except jwt.InvalidTokenError as token_error:
-            raise exceptions.AuthenticationFailed() from token_error
-        except Exception as ex:
-            # Errors in production do not need to be logged (as they may be noisy),
-            # but debug logging can help quickly resolve issues during development.
-            logger.debug(ex)
-            raise
+        except (AuthenticationFailed, PermissionDenied) as exception:
+            if authorization_header:
+                logger.debug('Invalid JWT auth header.', exc_info=exception)
+                set_custom_attribute('jwt_auth_failed', 'auth header:{}'.format(repr(exception)))
+                raise
+
+            # If we reach here, we must be using the JWT Cookie.
+            # For failed authentication using JWT Cookie, we will allow other Authentication classes
+            # to have a chance to attempt authentication by returning none, but will add logging just in case.
+            logger.debug('Invalid JWT Cookie. Proceeding to next Authentication class.', exc_info=exception)
+            set_custom_attribute('jwt_auth_failed', 'cookie:{}'.format(repr(exception)))
```

Comment:
@nedbat: `set_custom_attribute` mocking is not working, but `logger.debug` is, and `set_custom_attribute` mocking is working fine in `edx_rest_framework_extensions/auth/jwt/tests/test_middleware.py`, but not in the failing test from this file:
```
pytest edx_rest_framework_extensions/auth/jwt/tests/test_authentication.py::JwtAuthenticationTests::test_authenticate_csrf_protected
```

Reply:
This has been resolved.